### DISCUSSION TITLE
Provide/Distribute Mullvad VPN through Accrescent Store

### DISCUSSION THREAD

**Discussion by pxlkng** [mullvad/mullvadvpn-app#8925] [Feature requests & Ideas]
Currently the only direct way of getting Mullvad VPN on Android with automatic updates and avoiding Google is through the F-Droid store. This is not optimal and there are several [issues](https://privsec.dev/posts/android/f-droid-security-issues/) with this approach, most notably F-Droids really slow update time, their signing of all apks with their own key and therefore generally weak security model. More reasons are listed in the article.

Probably the best alternative currently available is [Accrescent](https://accrescent.app/), which does the concept of F-Droid right, with strong security mechanics like app signing key pinning, signed repository metadata, unprivileged automatic updates and developers own apk signing keys. (more also on their site)
It also has some other nice features like support for split-apks.

While I don't request that Mullvad stops to distribute Mullvad VPN through F-Droid, **I would like to suggest and strongly recommend also distributing Mullvad VPN through Accrescent** as at least another, safer/better, option and to maybe additionally recommend this way of installing over F-Droid for people that avoid Google Play.

On a more personal note I also feel like Accrescent is much more user-friendly and not as bloated with excess unneeded features like F-Droid. I think it goes well with Mullvads principle of being secure, private and easy to use in one.

Thank you for making Mullvad VPN and I hope that this could be considered.

**Comment by BHydden** [comment]
If this is your first time hearing about Accrescent, please be aware it is still in alpha and lacks many QOL features. It is also white list only currently so if you're interested discuss the matter with the dev otherwise just bookmark it for later when Accrescent is ready for widespread adoption.

**Comment by Rawa** [comment]
Hello, thanks for the feature request! 

We've had some internal discussions about deploying on Accrescent a few months back, and we have a ticket about evaluating if we should. The last few months we had some more pressing tasks but hopefully this is something we can do in the future.

**Comment by barbiehunter** [comment]
Are there any news ading Mullvad to Accrescent?

**Comment by Rawa** [comment]
No news as for right now I'm afraid. The feature request is still very much relevant and in our backlog on what we want to do. We've had other projects with more priority to take precedence. I don't want to give out an inaccurate timeline but the teams wish is that we are able to provide this in the near future.

**Comment by git70** [comment]
Hi guys!

I just wanted to add my +1!
Acrescent is gaining more and more attention in the privacy/security community.
It has also gained the approval of GrapheneOS and is included in their repository.
I love Mullvad, but the competition is already in Accrescent, so I can't imagine this store without you ;)

**Comment by ghost** [comment]
I would love to see you there too!

**Comment by ghost** [comment]
I agree, that we would love to see Mullvad VPN on Accrescent! Maybe you ask why we want this or what Benefits it would have for us as Users.

The Benefits for me would be a few, like these:

- I could get Mullvad VPN from Accrescent, which I already installed and use to download Apps. I prefer Accrescent over Github, because on Accrescent the App get's checked before it comes to the Store and the Apps are always checked, like for example the SDK of Apps, if there is Malware and more.

- It would make it a lot easier for me and save me time, because Mullvad VPN is one of the few Apps, I use in Obtainium to download. And if my last few Apps would come to Accrescent, I wouldn't need to use Obtainium. I just use it until my last few Apps on there come to Accrescent.

- Apps on Github are not checked, but on Accrescent they are going trough Filters always. And I think that would make it more secure. And since Accrescent is on a good way in terms of Security and Privacy, I think that it will be really good in the Future. And I think that Apps which value Security and Privacy should make it available there. Because Accrescent have the same values in Security and Privacy.

- I really hope that you Guys could make that possible, because it would have great Benefits for us as Users. And think like that: If we support them by adding Apps, we help them to grow from Alpha to Stable.

**Comment by ghost** [comment]
Is there any Updates on this?

**Comment by Pururun** [comment]
> Is there any Updates on this?

Not much. We want to do this, but we have prioritized other things such as multihop, DAITA and Shadowsocks for example.

**Comment by ghost** [comment]
> > Is there any Updates on this?
> 
> Not much. We want to do this, but we have prioritized other things such as multihop, DAITA and Shadowsocks for example.

Oh okay, I understand that, of course. Nice to know. Thank you for telling.

**Comment by boredsquirrel** [comment]
Just saying, Multihop seems to just have been added, and DAITA and Shadowsocks are there for a while ;)

**Comment by ghost** [comment]
I would also like to see mullvad on Accrescent, it's one of only 8 apps I currently need GPlay for.

Edit: it would also mean not having to wait for days until google finally approves updates.

**Comment by git70** [comment]
> it's one of only 8 apps I currently need GPlay for.

Alternatively, you can download directly from Releases https://github.com/mullvad/mullvadvpn-app/releases
and manage updates in Obtainium https://github.com/ImranR98/Obtainium

**Comment by ghost** [comment]
@git70 that would mean two things:
1. relying on Appverifier for verification (some apps I use are not in the internal database)
2. not having cert pinning and cert transparancy

I know about obtainium and specifically go out of my way to not use it, I only use secure app stores.

**Comment by Hasshu** [comment]
For what it's worth, a certain other provider's app has been [available](https://www.reddit.com/r/IVPN/comments/12stn5m/ivpn_for_android_is_now_available_on_accrescent/) over yonder for around two years (!) at this point.

**Comment by ghost** [comment]
It's been over a month since the stated goals have been achieved, could we please at least get a comment on whether this is still planned at all or not?

**Comment by cameronaaron** [comment]
I would also like to cast a vote for this

**Comment by ghost** [comment]
I would like it to be available on Accrescent as soon as the dev is available to accept app request too

**Comment by albin-mullvad** [comment]
Hi! Here's a small update: We have on our internal roadmap to evaluate and potentially add support for Accrescent later this year. The Android team is quite small and our focus in the build/release area have lately been [reproducible builds](https://mullvad.net/en/blog/reproducible-builds-verify-our-android-app-builds-bit-for-bit) as well as release automation which we've prioritized higher than adding another distribution channel. We also consider it to be more work than just uploading our binaries in another portal. We need to thoroughly evaluate Accrescent as a platform as well as build a robust automated release pipeline before we can start publishing there.

**Comment by git70** [comment]
@ghost
> 1. relying on Appverifier for verification (some apps I use are not in the internal database)
> 2. not having cert pinning and cert transparancy
>
> I know about obtainium and specifically go out of my way to not use it, I only use secure app stores.

1. If the application is not in the internal AppVerifier database, then you can / should check its signature by hand
2. The cert pinning is not made by the store, but the operating system (regardless of the source of the installation). He does it at the first installation, and later during the update he checks if the signature is the same.
3. Opinions on the safety of using stores vs direct download from the developer are divided. For example:
https://discuss.grapheneos.org/d/18731-f-droid-vulnerability-allows-bypassing-certificate-pinning
https://github.com/obfusk/fdroid-fakesigner-poc

@albin-mullvad 
This is a great news that Accrescent is on your road map!
Your explanation is proof that you are professionals and carefully examine every matter before its implementation.

**Comment by ghost** [comment]
Bumping this as well, I'd love to see Mullvad on Accrescent

**Comment by cameronaaron** [comment]
Any kind of update on this?

**Comment by Rawa** [comment]
@cameronaaron 

We've just set the plan for Q4, it includes looking into supporting more stores in October, with possible implementation in November. It is by no means set that we will support Accrescent, we've been hesitant since it is in alpha and the project halted October 2024 to June 2025., however it seems like development has started again. So exactly where this issue heads will be more clear in November.

**Comment by lberrymage** [comment]
> It is by no means set that we will support Accrescent, we've been hesitant since it is in alpha and the project halted October 2024 to June 2025

Accrescent developer here 👋🏻. Just wanted to add some clarification to this: Accrescent hasn't halted. We (Accrescent) said feature development would halt starting November 2025 if we didn't receive enough financial support to continue, but we quickly received enough support to continue through at least June 2026. Development is ongoing.

(edit: June 2026, not 2025)

**Comment by Rawa** [comment]
> > It is by no means set that we will support Accrescent, we've been hesitant since it is in alpha and the project halted October 2024 to June 2025
> 
> Accrescent developer here 👋🏻. Just wanted to add some clarification to this: Accrescent hasn't halted. We (Accrescent) said feature development would halt starting November 2025 if we didn't receive enough financial support to continue, but we quickly received enough support to continue through at least June 2026. Development is ongoing.
> 
> (edit: June 2026, not 2025)
> 

Hello, I think you misunderstood me, I'm talking about 2024 to 2025 June, by looking at git history. This feature request has been existed for about a year now. Im glad to hear you have funding moving forward. 👍 Ill get back and update this issue when we know more.

Best regards

**Comment by lberrymage** [comment]
Hi! Sorry about that; you're right, I misunderstood. I did want to come back and clarify though that the project didn't halt October 2024 to June 2025 either. There was a higher proportion of commits in private repositories (largely infrastructure work); larger, less frequent commits; more administration overhead work; more extensive design work; and a lot of long-term project planning in that period, which might explain why public git history looks sparse. There was, to be fair, about 1/4 of the amount of work going into Accrescent then as compared to now, but it didn't halt at any time.

No worries at all if that still causes hesitation, but I wanted to be sure to clarify what Accrescent was doing then. Cheers!

**Comment by albin-mullvad** [comment]
We've evaluated Accrescent as a potential distribution channel but have decided not to prioritize it at this time. Our decision is based on several considerations, but mainly on the lack of Accrescent support for release automation (CI/CD) ([issue for adding such support](https://github.com/accrescent/parcelo/issues/286)). This would require manual effort for each release, which we’re not willing to commit to at this point in time.

This isn't a definitive "no" – we'll continue monitoring Accrescent's development and may revisit this decision in the future if circumstances change.

We understand this may be disappointing for some users who prefer this platform, and we appreciate your understanding.
